bot/exts/fun/fun.py

```diff
@@ -80,6 +80,90 @@ def conversion_func(text: str) -> str:
             converted_text = f">>> {converted_text.lstrip('> ')}"
         await ctx.send(content=converted_text, embed=embed)
 
+    @commands.command(name="snakecase", aliases=("scase",))
+    async def snakecase_command(self, ctx: Context, *, text: str | None) -> None:
+        """Attempts to convert the provided string to snake_case."""
+        text = helpers.neutralise_string(text)
+        def conversion_func(text: str) -> str:
+            """Converts the provided string to snake_case."""
+            return "_".join(
+                text.split()
+            )
+        text = await clean_text_or_reply(ctx, text)
+        text, embed = await messages.get_text_and_embed(ctx, text)
+        # Convert embed if it exists
+        if embed is not None:
+            embed = messages.convert_embed(conversion_func, embed)
+        converted_text = conversion_func(text)
+        converted_text = helpers.suppress_links(converted_text)
+        # Don't put >>> if only embed present
+        if converted_text:
+            converted_text = f">>> {converted_text.lstrip('> ')}"
+        await ctx.send(content=converted_text, embed=embed)
+
+    @commands.command(name="pascalcase", aliases=("pcase", "pascal",))
+    async def pascalcase_command(self, ctx: Context, *, text: str | None) -> None:
+        """Attempts to convert the provided string to pascalCase."""
+        text = helpers.neutralise_string(text)
+        def conversion_func(text: str) -> str:
+            """Converts the provided string to pascalCase."""
+            return "".join(
+                word[0].upper()+word[1:] if i != 0 else word for i, word in enumerate(text.split())
+            )
+        text = await clean_text_or_reply(ctx, text)
+        text, embed = await messages.get_text_and_embed(ctx, text)
+        # Convert embed if it exists
+        if embed is not None:
+            embed = messages.convert_embed(conversion_func, embed)
+        converted_text = conversion_func(text)
+        converted_text = helpers.suppress_links(converted_text)
+        # Don't put >>> if only embed present
+        if converted_text:
+            converted_text = f">>> {converted_text.lstrip('> ')}"
+        await ctx.send(content=converted_text, embed=embed)
+
+    @commands.command(name="screamingsnakecase", aliases=("screamsnake", "ssnake", "screamingsnake",))
+    async def screamingsnakecase_command(self, ctx: Context, *, text: str | None) -> None:
+        """Attempts to convert the provided string to SCREAMING_SNAKE_CASE."""
```

Comment:
All these functions are following the exact same rules to convert the text, with the exception to the `conversion_func`.

This counts as them all duplicating the knowledge of text cleanup/conversion, which is something we don't want to have.

You could make a private function to the cog, that takes in the `ctxt, text, conversion_func` as parameters, and just call it in the `command` callbacks with the separate conversion definitions.

Reply:
Noted.
I've taken your feedback and extracted that bit of code into a function to then call the function.
I hope this is what you meant. :)
Any more feedback is appreciated.